File: src/main/java/de/darkatra/femtojar/FemtojarGradlePlugin.java
package de.darkatra.femtojar;

import org.gradle.api.Plugin;
import org.gradle.api.Project;
import org.gradle.api.tasks.TaskContainer;

@SuppressWarnings("unused") // used via build.gradle.kts
public class FemtojarGradlePlugin implements Plugin<Project> {

    @Override
    public void apply(final Project project) {

        final FemtojarExtension extension = project.getExtensions().create(
            "femtojar",
            FemtojarExtension.class
        );

        final TaskContainer tasks = project.getTasks();
        tasks.register("reencodeJars", ReencodeJarsTask.class, task -> {
            task.setGroup("femtojar");
            task.setDescription("Re-encode JAR files with custom class loader and compression.");

            // Configure task with extension values
            task.getSkip().set(extension.getSkip());
            task.getCompressionMode().set(extension.getCompressionMode());
            task.getBundleResources().set(extension.getBundleResources());
        });
    }
}

File: src/main/kotlin/de/darkatra/femtojar/FemtojarGradlePlugin.kt
package de.darkatra.femtojar

import org.gradle.api.Plugin
import org.gradle.api.Project

@Suppress("unused") // used via build.gradle.kts
class FemtojarGradlePlugin : Plugin<Project> {

    override fun apply(project: Project) {

        val extension = project.extensions.create("femtojar", FemtojarExtension::class.java)

        project.tasks.register("reencodeJar", ReencodeJarTask::class.java) { task ->
            task.group = "femtojar"
            task.description = "Re-encode JAR files with custom class loader and compression."

            task.skip.set(extension.skip)
            task.compressionMode.set(extension.compressionMode)
            task.bundleResources.set(extension.bundleResources)
            task.inputFile.set(extension.inputFile)
            task.outputFile.set(extension.outputFile)
        }
    }
}
